pkg/deployment/kube_service.go

```diff
@@ -34,18 +38,32 @@ func CreateKubeServices(
 	helper *helper.Helper,
 	labels map[string]string,
 ) error {
-	// We create one Service per port and per node, as the Service configuration states that if we
-	// just add all the compute nodes IPs to one service, the Service will round-robin between them.
-	// Our wanted behaviour is to expose all the compute nodes services at the same time.
+	// We create only one KubeService per port. All the nodes will be IPs on the endpointslices
+	// This will round-robin requests to the nodes, but it is also useful for Prometheus configuration
 	for _, kubeService := range instance.Spec.Services {
 		_, err := service(kubeService, instance, helper, labels)
 		if err != nil {
 			return err
 		}
 
+		var addressType discoveryv1.AddressType
 		addresses := make([]string, len(nodeSet.Spec.Nodes))
-		for _, item := range nodeSet.Spec.Nodes {
-			addresses = append(addresses, item.Ansible.AnsibleHost)
+		i := 0
+		for name, item := range nodeSet.Spec.Nodes {
+			namespacedName := &types.NamespacedName{
+				Name:      name,
+				Namespace: instance.GetNamespace(),
+			}
+			if len(item.Ansible.AnsibleHost) > 0 {
+				addresses[i], addressType = getAddressFromAnsibleHost(&item, namespacedName, helper)
```

Comment:
If set, using AnsibleHost always overrides using IPSet. Except when AnsibleHost is just a hostname (short), then an IPSet is never checked with this logic.

Reply:
Maybe we want to search first always in IPSets if we have them and resource to NodeSet only when those are not available?

Reply:
The ordering is ok. I don't have a strong opinion if one should take precedence over the other. But, if AnsibleHost is set, but you can't use it, then I'd suggest checking for IPSets at that point.

Reply:
I have rethinked this and I think that if we have IPSets and we allow the user to choose the network, it would be better to always look into IPSets first, as that option allows us to choose from the user's chosen network. If we do not have IPSets, then the network choice from the user is not important and we just will try to expose the AnsibleHost, as a backup.